flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnResourceManagerFactory.java

```diff
@@ -81,4 +91,27 @@ public static YarnResourceManagerFactory getInstance() {
 			webInterfaceUrl,
 			resourceManagerMetricGroup);
 	}
+
+	@Override
+	protected CPUResource getDefaultCpus(final Configuration configuration) {
+		int fallback = configuration.getInteger(YarnConfigOptions.VCORES);
+		double cpuCoresDouble = TaskExecutorProcessUtils.getCpuCoresWithFallback(configuration, fallback).getValue().doubleValue();
+		@SuppressWarnings("NumericCastThatLosesPrecision")
+		long cpuCoresLong = Math.max((long) Math.ceil(cpuCoresDouble), 1L);
+		//noinspection FloatingPointEquality
+		if (cpuCoresLong != cpuCoresDouble) {
+			LOG.info(
+				"The amount of cpu cores must be a positive integer on Yarn. Rounding {} up to the closest positive integer {}.",
+				cpuCoresDouble,
+				cpuCoresLong);
+		}
+		if (cpuCoresLong > Integer.MAX_VALUE) {
```

Comment:
I think it is not robust to make the correctness of `YarnResourceManagerFactory` depending on another class `YarnClusterDescriptor`. There is no harm for double checking it, right?

Reply:
A precondition should work as defensive programming.

Reply:
@TisonKun Not sure what do you mean. Are you suggesting simplify this code with a `Preconditions.checkstate()`?
I remember the reason we implemented it like this in `YarnResourceManager` at the first place, is to get an `IllegalConfigurationException` instead of a `IllegalStateException`, for better readability to the users.

Reply:
@xintongsong that makes sense.

Reply:
I am not against with the check. I just feel the `Integer.MAX_VALUE` checking will not take too much effect. Since the `YarnMaxVcores` is far less than `Integer.MAX_VALUE`.

Centainly, you could keep the current check. It is safer when converting `long` to `int`.